### PR TITLE
[Merged by Bors] - chore(Calculus/Inverse): split

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Calculus/FDeriv/Equiv.lean [hunk 1]
```diff
@@ -16,8 +16,10 @@ see the module docstring of `Analysis/Calculus/FDeriv/Basic.lean`.
 
 This file contains the usual formulas (and existence assertions) for the derivative of
 continuous linear equivalences.
--/
 
+We also prove the usual formula for the derivative of the inverse function, assuming that it exists.
```
Comment (grunweg) [inline Mathlib/Analysis/Calculus/FDeriv/Equiv.lean:20]:
```suggestion
We also prove the usual formula for the derivative of the inverse function, assuming it exists.

File: Mathlib/Analysis/Calculus/Inverse/FDeriv.lean [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+open Asymptotics Filter Metric Set
+
+open ContinuousLinearMap (id)
+
+
+/-!
+### Inverse function theorem
+
+Now we prove the inverse function theorem. Let `f : E → F` be a map defined on a complete vector
```
Comment (grunweg) [inline Mathlib/Analysis/Calculus/Inverse/FDeriv.lean:69]:
```suggestion
Let `f : E → F` be a map defined on a complete vector

File: Mathlib/Analysis/Calculus/Inverse/FiniteDimensional.lean [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.Calculus.Inverse.ApproximatesLinearOn
+import Mathlib.Analysis.NormedSpace.FiniteDimension
+
+/-!
+# A lemma about `ApproximatesLinearOn` that needs `FiniteDimensional`
+
+In this file we prove that In a real vector space,
```
Comment (grunweg) [inline Mathlib/Analysis/Calculus/Inverse/FiniteDimensional.lean:12]: capitalisation :-)

**Review by grunweg — COMMENTED** [review]
Thanks for splitting this large file! The other day I noticed that this file was getting slow, and now you just went ahead and did the split :-)

The new grouping makes sense to me; I wonder whether
- these files need to be split already? The original file was just at 900 lines... or is the point merely that they're four to five different things, so splitting makes more sense? (I find the latter argument convincing enough.)
- the several small files pull their weight.

Or do you intend to add further material to this file, and this is just anticipatory clean-up?
Presumably, you've thought about this; please enlighten me :-)

**Comment by urkud** [comment]
The main reason for me was moving parts with different dependencies to different files. Different maintainers have different preferences about the optimal size of a file. I have a slow laptop, so I prefer short files. Also, I'm going to expand the `ContDiff` file later.

File: Mathlib/Analysis/Calculus/Inverse/FiniteDimensional.lean (hunk 3 above)
Comment (urkud) [reply Mathlib/Analysis/Calculus/Inverse/FiniteDimensional.lean:12]:
```suggestion
In this file we prove that in a real vector space,
```

**Review by grunweg — APPROVED** [review]
Thanks for the quick response! Your explanation makes sense to me; I'm convinced.

File: Mathlib/Analysis/Calculus/FDeriv/Mul.lean [hunk 4]
```diff
@@ -17,8 +17,7 @@ This file contains the usual formulas (and existence assertions) for the derivat
 
 * multiplication of a function by a scalar function
 * multiplication of two scalar functions
-* inverse function (assuming that it exists; the inverse function theorem is in
-  `Mathlib/Analysis/Calculus/Inverse.lean`)
+* inverse function (as in `Ring.inverse` or `Inv.inv`, not `Function.LeftInverse`).
```
Comment (hrmacbeth) [inline Mathlib/Analysis/Calculus/FDeriv/Mul.lean:20]:
```suggestion
* taking the pointwise multiplicative inverse (i.e. `Inv.inv` or `Ring.inverse`) of a function
```

**Review by hrmacbeth — COMMENTED** [review]
This looks good!

I think it would be cleared to change the folder name from `Inverse` to `InverseFunctionTheorem`, what do you think?

bors d+

File: Mathlib/Analysis/Calculus/Inverse/Deriv.lean [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+-/
+import Mathlib.Analysis.Calculus.Deriv.Inverse
+import Mathlib.Analysis.Calculus.Inverse.FDeriv
+
+/-!
+# Inverse function theorem, 1D case
+
+In this file we prove a version of the inverse function theorem for maps `f : 𝕜 → 𝕜`.
+We use `ContinuousLinearEquiv.unitsEquivAut` to translate `HasStrictDerivAt f f' a` and
+`f' ≠ 0` into `HasStrictFDerivAt f (_ : 𝕜 ≃L[𝕜] 𝕜) a`.
```
[the remark this answers is not in the document]
Comment (urkud) [reply Mathlib/Analysis/Calculus/Inverse/Deriv.lean]: This needs much more than `HasStrictDerivAt` at a point. Could you please suggest a specific wording (or just push it to the branch)?

**Comment by urkud** [comment]
I'm merging this PR as docstring improvements can be dealt with later.
bors merge

**Pull request closed without merging**